istio-control/istio-autoinject/files/injection-template.yaml

```diff
@@ -115,10 +117,12 @@ template: |
     - --datadogAgentAddress
     - "{{ .ProxyConfig.GetTracing.GetDatadog.GetAddress }}"
   {{- end }}
-    - --proxyLogLevel
-    - "{{ annotation .ObjectMeta `sidecar.istio.io/debug` .Values.global.proxy.logLevel}}"
-    - --log_output_level
-    - '{{ .Values.debug }}'
+  {{- if .Values.global.proxy.logLevel }}
+    - --proxyLogLevel={{ .Values.global.proxy.logLevel }}
+  {{- end}}
+  {{- if .Values.global.proxy.componentLogLevel }}
```

Comment:
I would follow the same pattern - log level at a global mesh level is not very useful.

Reply:
Honestly not really sure why you would want to change the log level for a single pod. If you are just debugging you can change it at runtime in Envoy.

Reply:
Runtime change implies kubectl exec to the machine, running some commands - and if envoy crashes or restarts doing it again. 

I don't want to do it at pod level - but deployment level. For example the staging or test workloads will have it set to debug, while prod to warn.  

The global setting for any control plane that has prod jobs can really only be 'debug off' - I wouldn't mind removing the global and only supporting workload-specific settings.

This happens to be a prefect example for the new patching API and kustomize.